frontend/components/Editor.js

```diff
@@ -1046,6 +1058,27 @@ patch: ${JSON.stringify(
             }
         }
 
+        this.desktop_submit_file_change = async () => {
+            window.plutoDesktop?.ipcRenderer.once("PLUTO-MOVE-NOTEBOOK", async (/** @type {string | undefined} */ loc) => {
+                try {
+                    if (!!loc)
+                        await update_notebook((notebook) => {
+                            notebook.in_temp_dir = false
+                            notebook.path = loc
+                        }, false)
```

Comment:
Can you change this to the following, so that we don't need to modify the `update_notebook` function?

```js
await this.setStatePromise(
    immer((notebook) => {
        notebook.in_temp_dir = false
        notebook.path = loc
    })
)
```

Reply:
This one is not only related to changing the state, I need the extra parameter for what is happening on line [990](https://github.com/Illusion47586/Pluto.jl/blob/f12e07063a44e61c4909a34bce7428c53687bf99/frontend/components/Editor.js#L990)

Reply:
You need it for what exactly? If line 990 is false, then it will run the snippet from my comment, plus [this](https://github.com/Illusion47586/Pluto.jl/blob/f12e07063a44e61c4909a34bce7428c53687bf99/frontend/components/Editor.js#L944-L985), but I don't think you need it. Does it not work otherwise?

Reply:
Tried it and it is working! pushed it.